websocket/polygon.go

```diff
@@ -313,6 +314,7 @@ func (c *Client) write() error {
 	defer func() {
 		c.log.Debugf("write thread closed")
 		ticker.Stop()
+		c.authenticated = false
```

Comment:
Can you explain the changes to the `authenticated` flag? I don't see its relevance to processing rawData.

Reply:
Nevermind I understand its functionality now.